core/runtime/src/main/java/io/quarkus/runtime/JVMChecksRecorder.java

```diff
@@ -0,0 +1,31 @@
+package io.quarkus.runtime;
+
+import java.lang.management.ManagementFactory;
+import java.lang.management.RuntimeMXBean;
+import java.util.List;
+
+import org.jboss.logging.Logger;
+
+import io.quarkus.runtime.annotations.Recorder;
+
+@Recorder
+public class JVMChecksRecorder {
+
+    public void check() {
+        if (!isUnsafeMemoryAccessAllowed()) {
+            Logger.getLogger("JVM").warn(
+                    "Unsafe memory access is not allowed by default on Java since version 24, but several Quarkus extensions still require it. "
+                            +
+                            "Please add the `--sun-misc-unsafe-memory-access=allow` JVM argument to retain the semantics of older JVMs. "
+                            +
+                            "We are working with the maintainers of those libraries to get this resolved in future versions.");
+        }
+    }
+
+    public static boolean isUnsafeMemoryAccessAllowed() {
+        RuntimeMXBean runtimeMxBean = ManagementFactory.getRuntimeMXBean();
+        List<String> arguments = runtimeMxBean.getInputArguments();
+        return arguments.contains("--sun-misc-unsafe-memory-access=allow");
+    }
```

Comment:
Will this even work in native mode?

Reply:
good point. I'll disable it for native as we control the native-image creation commandline